pkgs/desktops/plasma-5/plasma-desktop/default.nix

```diff
@@ -122,19 +129,20 @@ mkDerivation {
     ./kcm-access.patch
   ];
   CXXFLAGS =
-    let
-      # run gsettings with desktop schemas for using in kcm_accces kcm
-      gsettings-wrapper = runCommandLocal "gsettings-wrapper" { nativeBuildInputs = [ makeWrapper ]; } ''
-        makeWrapper ${glib}/bin/gsettings $out --prefix XDG_DATA_DIRS : ${gsettings-desktop-schemas.out}/share/gsettings-schemas/${gsettings-desktop-schemas.name}
-      '';
-    in
     [
       ''-DNIXPKGS_HWCLOCK=\"${lib.getBin util-linux}/bin/hwclock\"''
-      ''-DNIXPKGS_GSETTINGS=\"${gsettings-wrapper}\"''
+      ''-DNIXPKGS_GSETTINGS=\"${gsettings-wrapper}/bin/gsettings\"''
     ];
   postInstall = ''
     # Display ~/Desktop contents on the desktop by default.
     sed -i "''${!outputBin}/share/plasma/shells/org.kde.plasma.desktop/contents/defaults" \
         -e 's/Containment=org.kde.desktopcontainment/Containment=org.kde.plasma.folder/'
   '';
+  postFixup = ''
+    # put in postFixup for preventing wrapQtAppsHook wrapping wrapper wrapper
+
+    # wrap kacces with wrapped gsettings so it can access accessibility schemas
+    # double wrapping kaccess here because wrapping everything in plasma-desktop with qtWrapperArgs would be overkill
```

Comment:
We should avoid double wrapping at all costs

Reply:
is there any other reason to avoid double wrapping than it is being hacky and adding unnecessary indirection ?

Reply:
argv0 is often wrong at the second layer of wrapping